sdk/src/main/scala/io/horizen/account/state/ForgerStakeV2MsgProcessor.scala

```diff
@@ -0,0 +1,588 @@
+    val cmdInput = StakeTotalCmdInputDecoder.decode(inputParams)
+
+    val forgerKeys = cmdInput.forgerPublicKeys
+    val delegator = cmdInput.delegator
+    val consensusEpochStart = if (cmdInput.consensusEpochStart.isEmpty) currentEpoch else cmdInput.consensusEpochStart.get
+    val maxNumOfEpoch = cmdInput.maxNumOfEpoch
+    log.info(s"stakeTotal called - $forgerKeys $delegator epochStart: $consensusEpochStart - maxNumOfEpoch: $maxNumOfEpoch")
+
+    if (forgerKeys.isEmpty && delegator.isDefined) {
+      val msgStr = s"Illegal argument - delegator is defined while forger keys are not"
+      throw new ExecutionRevertedException(msgStr)
+    }
+    val consensusEpochEnd =
+      if (maxNumOfEpoch.isEmpty) consensusEpochStart
+      else if (consensusEpochStart + maxNumOfEpoch.get > currentEpoch) currentEpoch
+      else consensusEpochStart + maxNumOfEpoch.get - 1
+
+    val response: StakeTotalCmdOutput = StakeStorage.getStakeTotal(view, forgerKeys, delegator, consensusEpochStart, consensusEpochEnd)
+
+    response.encode()
+  }
+
+  def doRewardsReceivedCmd(invocation: Invocation, view: BaseAccountStateView, metadata: MsgProcessorMetadataStorageReader, currentEpoch: Int): Array[Byte] = {
+    requireIsNotPayable(invocation)
+    checkForgerStakesV2IsActive(view)
+
+    val inputParams = getArgumentsFromData(invocation.input)
+    val cmdInput = RewardsReceivedCmdInputDecoder.decode(inputParams)
+
+    val forgerKeys: ForgerPublicKeys = cmdInput.forgerPublicKeys
+    val consensusEpochStart: Int = cmdInput.consensusEpochStart
+    val maxNumOfEpoch: Int =
+      if (consensusEpochStart + cmdInput.maxNumOfEpoch - 1 >= currentEpoch)
+        currentEpoch - consensusEpochStart
+      else
+        cmdInput.maxNumOfEpoch
+
+    log.info(s"rewardsReceived called - $forgerKeys epochStart: $consensusEpochStart")
+    StakeStorage.getForger(view, forgerKeys.blockSignPublicKey, forgerKeys.vrfPublicKey)
+      .getOrElse(throw new ExecutionRevertedException("Forger doesn't exist."))
+
+    val rewards: Seq[BigInteger] = metadata.getForgerRewards(forgerKeys, consensusEpochStart, maxNumOfEpoch)
+    val response = RewardsReceivedCmdOutput(rewards)
+
+    response.encode()
+  }
+
+  def doStakeStartCmd(invocation: Invocation, view: BaseAccountStateView): Array[Byte] = {
+    requireIsNotPayable(invocation)
+    checkForgerStakesV2IsActive(view)
+
+    val inputParams = getArgumentsFromData(invocation.input)
+    val cmdInput = StakeStartCmdInputDecoder.decode(inputParams)
+
+    val forgerKeys = cmdInput.forgerPublicKeys
+    val delegator = cmdInput.delegator
+    log.info(s"stakeStart called - $forgerKeys $delegator")
+
+    val response: StakeStartCmdOutput = StakeStorage.getStakeStart(view, forgerKeys, delegator)
+    response.encode()
+  }
+
+  def doPagedForgersStakesByDelegatorCmd(invocation: Invocation, view: BaseAccountStateView): Array[Byte] = {
+    requireIsNotPayable(invocation)
+    if (!StakeStorage.isActive(view)) {
```

Comment:
We had better use `checkForgerStakesV2IsActive(view)` method to avoid code duplication. The same applies for other methods below.